packages/playground/website/src/components/ensure-playground-site/index.tsx

```diff
@@ -0,0 +1,16 @@
+import { EnsurePlaygroundSiteIsSelected } from './ensure-playground-site-is-selected';
+import { EnsurePlaygroundSiteSlug } from './ensure-playground-site-slug';
+
+export function EnsurePlaygroundSite({
+	children,
+}: {
+	children: React.ReactNode;
+}) {
+	return (
+		<EnsurePlaygroundSiteSlug>
+			<EnsurePlaygroundSiteIsSelected>
+				{children}
```

Comment:
This seems like a really elegant model. You wrap the inner UI components with explicit assumption components. How cool.